django/contrib/admin/sites.py

```diff
@@ -533,16 +531,17 @@ def index(self, request, extra_context=None):
 
         return TemplateResponse(request, self.index_template or 'admin/index.html', context)
 
+    @never_cache
     def app_index(self, request, app_label, extra_context=None):
-        app_dict = self._build_app_dict(request, app_label)
-        if not app_dict:
+        app_list = self.get_app_list(request, app_label)
+
+        if not app_list:
             raise Http404('The requested admin page does not exist.')
-        # Sort the models alphabetically within each app.
-        app_dict['models'].sort(key=lambda x: x['name'])
+
         context = {
             **self.each_context(request),
```

Comment:
`self.each_context` actually already contains a fully populated app list, under `available_apps`. We could make this more efficient by extracting `app_list` from `available_apps` rather than calculating it twice. 
```
context = self.each_context(request)
app_list = context['available_apps'].get(app_label)
if not app_list:
    raise Http404('The requested admin page does not exist.')
context = {
    **context
    ...
}
```